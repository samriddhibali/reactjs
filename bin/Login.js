import React, { Component } from 'react';

class Login extends Component {
    render(){
        return(
        <div className = 'Login'>
            <h1>Login Page</h1>
            <form name = "login" method="get" onSubmit = {this.props.onNavigate}>
            <label>Enter Username: </label>
            <input type="text" name="username" required/><br/><br/>
            <label>Enter Password: </label>
            <input type="password" name="password" required/><br/><br/>
            <input type = "submit" name="submit"/>
            </form>
        </div>
        );
    }
}

export default Login;