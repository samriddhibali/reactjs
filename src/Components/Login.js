import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from '../../../../../Library/Caches/typescript/3.0/node_modules/redux';
import { validateLogin } from '../ActionCreator/LoginAction';

class Login extends Component {

    constructor(props) {
        super(props);
        this.state = {
            uname: '',
            password: ''
        };
    }

    handleInput = (e) => {
        this.setState({
            [e.target.name]: e.target.value
        });
    }

    validateData = () => {
        const user = {
            "uname": this.state.uname.trim(),
            "password": this.state.password.trim()
        }
        this.props.validateLogin(user)
            .then(() => {
                if (this.props.data.newUser && this.props.data.status) {
                    alert("Now enter your details")
                    window.location.replace('/Form')
                }
                else {
                    this.props.history.go(0);
                }
            })
            .catch(() => {
                this.props.history.go(0);
            })
    }

    checkValid = () => {
        if (this.props.data.status) {
            localStorage.setItem("logged", 'success');
            // this.props.history.go(0);
        }
    }

    render() {
        this.checkValid();
        console.log("Data---           ", this.props.data)

        return (
            <div className = "Login">
                <h1>Login</h1>
                {
                    this.props.data.msg &&
                    <i>{this.props.data.msg}</i>
                }
                {/* <form className = "Signup" method="post"> */}
                <div className = "field">

                    <label>Username: </label>
                    <input type="text" name="uname" className="userName" onChange={this.handleInput} /><br /><br /><br />

                    <label>Password: </label>
                    <input type="password" name="password" className="password" onChange={this.handleInput} /><br /><br /><br />

                    <button className='button' onClick={this.validateData}>Submit</button>&nbsp;&nbsp;&nbsp;&nbsp;
                        <Link to="/Signup" ><button className="button add">New User</button></Link>
                </div>
                {/* </form> */}
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        data: state.LoginReducer,
    //    newUser: state.LoginReducer.newUser,
    //    status: state.LoginReducer.status,
    }
}

const mapDispatchToProps = (dispatch) => {
    console.log("dispatchToProps");
    return bindActionCreators({
        validateLogin: validateLogin,
    }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Login));