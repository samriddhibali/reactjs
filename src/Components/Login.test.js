import React from 'react';
import { withRouter } from 'react-router-dom';
import { configure, render, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import expect from 'expect';
import Login from './Login';
import renderer from 'react-test-renderer';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

describe('<Login/> Component', () => {
    const initialState = {
        LoginReducer:
        {
            msg: '',
            status: true,
            newUser: true
        }
    }; 
     const mockStore = configureStore();
    //  const createStore = configureStore();
     let store ;
    //  let actualStore;

     beforeEach(() => {
        store = mockStore(initialState);
      //  actualStore = createStore(LoginReducer);
    });
   
    test('should contain Login class with snapshot', () => {
        const wrapper = shallow(<Login store = {store}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('should contain input field with password classname', () => {
        const wrapper = shallow(<Login store = {store}/>);
        expect(wrapper.exists('input.password'));
    });

    test('checking props', () => {
        const wrapper = shallow(<Login store = {store}/>);
      // wrapper.setProps({newUser:true});
        console.log("wrapper debug  ",wrapper.props())
        expect(wrapper.prop('data').newUser).toEqual(initialState.LoginReducer.newUser);
    });

});