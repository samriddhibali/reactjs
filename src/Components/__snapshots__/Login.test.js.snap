// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> Component should contain Login class with snapshot 1`] = `
<withRouter(Login)
  data={
    Object {
      "msg": "",
      "newUser": true,
      "status": true,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  validateLogin={[Function]}
/>
`;
